### PR TITLE
Deprecate standalone SQL store level

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
This PR aims to eliminate database interaction code for standalone and single-user applications.

Stack on: https://github.com/mattermost/mattermost-plugin-boards/pull/2

### CODE REVIEW

File: server/services/store/sqlstore/board.go [hunk 1]
```diff
@@ -234,36 +234,6 @@ func (s *SQLStore) getBoard(db sq.BaseRunner, boardID string) (*model.Board, err
 	return s.getBoardByCondition(db, sq.Eq{"id": boardID})
 }
 
-func (s *SQLStore) getBoardsForUserAndTeam(db sq.BaseRunner, userID, teamID string, includePublicBoards bool) ([]*model.Board, error) {
```
Comment (harshilsharma63) [inline server/services/store/sqlstore/board.go:237]: Once this is removed, we can get rid of the `mattermostauthlayer` dir by bring its code over here. Up to you if you wanna handle that in this PR or create a separate PR for it. If handling in a separate PR, can you please create a Jira ticket to track it?

File: server/services/store/sqlstore/public_methods.go [hunk 2]
```diff
@@ -47,12 +48,12 @@ func (s *SQLStore) AddUpdateCategoryBoard(userID string, categoryID string, boar
 }
 
 func (s *SQLStore) CanSeeUser(seerID string, seenID string) (bool, error) {
```
Comment (harshilsharma63) [inline server/services/store/sqlstore/public_methods.go:50]: Why not delete this function as its implemented in `mattermostauthlayer`?

File: server/services/store/sqlstore/public_methods.go [hunk 3]
```diff
@@ -47,12 +48,12 @@ func (s *SQLStore) AddUpdateCategoryBoard(userID string, categoryID string, boar
...
-	return s.canSeeUser(s.db, seerID, seenID)
+	return false, errors.New("can see user not supported in focalboard, will fetch from mattermost") 
 
 }
 
 func (s *SQLStore) CleanUpSessions(expireTime int64) error {
```
Comment (harshilsharma63) [inline server/services/store/sqlstore/public_methods.go:55]: These session function should be deleted from here, from `mattermostauthlayer` as well as they are no longer needed anywhere in the app.

File: server/services/store/sqlstore/public_methods.go [hunk 4]
```diff
@@ -139,7 +140,7 @@ func (s *SQLStore) CreateSubscription(sub *model.Subscription) (*model.Subscript
 }
 
 func (s *SQLStore) CreateUser(user *model.User) (*model.User, error) {
```
Comment (harshilsharma63) [inline server/services/store/sqlstore/public_methods.go:142]: Why not simply delete these function where you are directly returning an error?

**Review by harshilsharma63 — CHANGES_REQUESTED** [review]

File: server/services/store/sqlstore/public_methods.go (hunk 3 above)
Comment (Rajat-Dabade) [reply server/services/store/sqlstore/public_methods.go]: Sure.

File: server/services/store/sqlstore/public_methods.go (hunk 4 above)
Comment (Rajat-Dabade) [reply server/services/store/sqlstore/public_methods.go]: Deleted.

File: server/services/store/sqlstore/board.go (hunk 1 above)
Comment (Rajat-Dabade) [reply server/services/store/sqlstore/board.go:237]: I will take care of this in subsequent PR.
Created a ticket for this: https://mattermost.atlassian.net/browse/MM-59349

File: server/services/store/sqlstore/public_methods.go (hunk 2 above)
Comment (Rajat-Dabade) [reply server/services/store/sqlstore/public_methods.go:50]: This required more cleaner, will do this as a part of this ticket: https://mattermost.atlassian.net/browse/MM-59349

File: server/services/store/sqlstore/public_methods.go (hunk 3 above)
Comment (Rajat-Dabade) [reply server/services/store/sqlstore/public_methods.go]: Removed it.

File: server/auth/auth_test.go [hunk 5]
```diff
@@ -54,37 +54,9 @@ func setupTestHelper(t *testing.T) *TestHelper {
 
 func TestGetSession(t *testing.T) {
```
Comment (harshilsharma63) [inline server/auth/auth_test.go:55]: There is nothing in this test left now, so we can delete this.

**Review by harshilsharma63 — APPROVED** [review]
Approving with one minor comment.

File: server/auth/auth_test.go (hunk 5 above)
Comment (Rajat-Dabade) [reply server/auth/auth_test.go]: Removed it.

**Review by sbishel — APPROVED** [review]
Just some comments regarding another variable, I think can be removed. And it looks like there is already plans to completely remove `mattermostauthlayer.go`.

File: server/auth/auth_test.go [hunk 6]
```diff
@@ -22,71 +13,6 @@ type TestHelper struct {
...
-			if test.isError {
-				require.Error(t, err)
-			} else {
-				require.NoError(t, err)
-				require.NotNil(t, session)
-			}
-		})
-	}
-}
-
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply server/auth/auth_test.go:89]: Created an issue for this: https://github.com/mattermost/mattermost-plugin-boards/issues/9

File: server/server/initHandlers.go [hunk 7]
```diff
@@ -1,6 +1,5 @@
 package server
 
 func (s *Server) initHandlers() {
-	cfg := s.config
-	s.api.MattermostAuth = cfg.AuthMode == MattermostAuthMod
+	s.api.MattermostAuth = true
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply server/server/initHandlers.go]: Removed the file itself as it was no long need.

File: server/api/auth.go [hunk 8]
```diff
@@ -5,20 +5,15 @@ import (
...
-	return a.attachSession(handler, true)
+	return a.attachSession(handler)
 }
 
-func (a *API) attachSession(handler func(w http.ResponseWriter, r *http.Request), required bool) func(w http.ResponseWriter, r *http.Request) {
+func (a *API) attachSession(handler func(w http.ResponseWriter, r *http.Request)) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
-		token, _ := auth.ParseAuthTokenFromRequest(r)
-
 		if a.MattermostAuth && r.Header.Get("Mattermost-User-Id") != "" {
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply server/api/auth.go]: Yup, that's a nice catch. Removed it.

File: server/api/archive.go [hunk 9]
```diff
@@ -204,51 +204,7 @@ func (a *API) handleArchiveExportTeam(w http.ResponseWriter, r *http.Request) {
...
-	opts := model.ExportArchiveOptions{
-		TeamID:   teamID,
-		BoardIDs: ids,
-	}
-
-	filename := fmt.Sprintf("archive-%s%s", time.Now().Format("2006-01-02"), archiveExtension)
-	w.Header().Set("Content-Type", "application/octet-stream")
-	w.Header().Set("Content-Disposition", "attachment; filename="+filename)
-	w.Header().Set("Content-Transfer-Encoding", "binary")
+	a.errorResponse(w, r, model.NewErrNotImplemented("not permitted in plugin mode"))
```
Comment (Rajat-Dabade) [inline server/api/archive.go:207]: @sbishel, as we only return an error response from this function after removing the `MattermostAuth` flag. Will this be a good idea to remove the full API end-point? Thoughts?

**Pull request merged**